sdk/orm-django/django_encryption/fields.py

```diff
@@ -42,8 +44,163 @@ class _RaiseError:
 class WithVaultOptions(Options):
     vault_collection: Optional[str] = None
 
+# This function is necessary so that we are able to pass information from the queryset
+# to get_prefetch_queryset
+
+
+def make_iterable_wrapper(transform_fields=None):
+    class ModelIterableWrapper(django.db.models.query.ModelIterable):
+
+        def __iter__(self):
+            for obj in super().__iter__():
+                if self.transform_fields:
+                    obj._transform_fields = self.transform_fields
+                yield obj
+    ModelIterableWrapper.transform_fields = transform_fields
+    return ModelIterableWrapper
+
+# this QuerySet subclass allows us to support transformations as part of the query
+# e.g. queryset.mask('field1', 'field2')
+
+
+class EncryptionBatchQuerySet(django.db.models.query.QuerySet):
+
+    def __init__(self, model=None, query=None, using=None, hints=None):
+        super().__init__(model, query, using, hints)
+        self._transform_fields = {}
+
+    def transform(self, transformation_name, *fields):
+        clone = self._clone()
```

Comment:
Because this is how querysets work in django, you always return a clone copy. 
No function is in-place. 
So you do something like

query = MyModel.objects.filter(field=value).mask(MyModel.field2)
query2 = query.filter(field3=value3)

and query is not changed by the second line defining query2.